.circleci/osx_install_dependencies.sh

```diff
@@ -54,33 +54,65 @@ if [ ! -f /usr/local/lib/libz3.a ] # if this file does not exists (cache was not
 then
   brew update
   brew upgrade
-  brew install boost
   brew install cmake
   brew install wget
   brew install coreutils
   brew install diffutils
   brew install grep
-  ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+
+  # writing to /usr/local/lib need administrative privileges.
+  sudo ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+
+  # boost
+  boost_version="1.84.0"
+  boost_package="boost_${boost_version//./_}.tar.bz2"
+  boost_dir="boost_${boost_version//./_}"
+  wget "https://boostorg.jfrog.io/artifactory/main/release/$boost_version/source/$boost_package"
+  tar xf "$boost_package"
+  rm "$boost_package"
+  cd "$boost_dir"
+  ./bootstrap.sh --with-toolset=clang --with-libraries=thread,system,filesystem,program_options,serialization,test
+  sudo ./b2 -a address-model=64 architecture=arm+x86 install
```

Comment:
Does `brew install boost` only provide an x86 build, i.e. is that why we're building boost from source?

Reply:
I removed the brew installation of boost. The thing is that we need the boost libraries in universal binary format, thats why we compile boost from source (arm+x86).